src/sparseml/exporters/kv_cache_injector.py

```diff
@@ -43,44 +43,82 @@ def __init__(
         autoregressive generation process (reduce the compute of key/value pairs
         by storing the results of previous computations in memory).
 
+        The exporter will look for a `config.json` file in the `model_path` directory
+        to determine the static dimensions of the kv cache input/output.
+
         This transformation not only injects the cache support, but also adjusts
         the model to account for the cache support. This means altering the input
         to the model, such as adding "position" input to the model.
 
         Usage:
         ```python
         onnx_model: onnx.ModelProto = ...
-        exporter = KeyValueCacheInjector()
+        exporter = KeyValueCacheInjector(model_path="path/to/model")
         exporter.export(onnx_model, "model.onnx")
         ```
 
         You can also just optimize the model directly without saving to disk:
         ```python
         onnx_model: onnx.ModelProto = ...
-        exporter = KeyValueCacheInjector()
+        exporter = KeyValueCacheInjector(model_path="path/to/model")
         optimized_model = exporter.apply(onnx_model)
         ```
 
-        :param model_type: The type of model to inject the cache support into.
+        :param model_path: The path to the directory containing the model.
         :param inplace: If True, the model will be modified in place.
             If False, a copy of the model will be made and modified.
         """
         self.inplace = inplace
-        if model_type.lower() not in _SUPPORTED_ARCHITECTURES:
+        self.config = self.get_config(model_path)
+
+        if not self.config["model_type"] in _SUPPORTED_ARCHITECTURES:
             raise ValueError(
-                f"`model_type` must be one of {_SUPPORTED_ARCHITECTURES}, "
```

Comment:
let's instead warn and just set to a dynamic value so as to not block users who want to try new models

Reply:
Good idea.